internal/dinosql/config.go

```diff
@@ -58,43 +59,78 @@ type Override struct {
 	GoType string `json:"go_type"`
 
 	// fully qualified name of the Go type, e.g. `github.com/segmentio/ksuid.KSUID`
-	PostgresType string `json:"postgres_type"`
+	DBType                  string `json:"db_type"`
+	Deprecated_PostgresType string `json:"postgres_type"`
+
+	// for global overrides only when two different engines are in use
+	Engine Engine `json:"engine,omitempty"`
 
 	// True if the GoType should override if the maching postgres type is nullable
 	Null bool `json:"null"`
 
 	// fully qualified name of the column, e.g. `accounts.id`
 	Column string `json:"column"`
 
-	columnName  string
-	table       pg.FQN
-	goTypeName  string
+	ColumnName  string
+	Table       pg.FQN
+	GoTypeName  string
 	goPackage   string
 	goBasicType bool
 }
 
+func (c *GenerateSettings) GatherEngines() map[Engine]struct{} {
```

Comment:
This function is only used in a single spot. Can you please inline into the `ValidateGlobalOverrides` function? I like to only break out functions into separate pieces when needed.